### PR TITLE
Update DPoS Unlocks endpoint

### LINKED ISSUE #1350: Update DPoS Unlocks endpoint
### Description

Update the following endpoint.

*Current:*
```
HTTP GET /api/v3/dpos/unlocks
RPC  get.dpos.unlocks
```

*Expected:*
```
HTTP GET /api/v3/pos/unlocks
RPC  get.pos.unlocks
```

### Acceptance Criteria

- The endpoint are updated as expected above
- The response body is up-to-date
- All the methods, variables, files are renamed and refactored appropriately
- Swagger is up-to-date
- Integration tests are up-to-date

### Additional information
*Request query param:*
- Add support for: `isUnlockable: boolean (optional)`

*Response:*

200 OK
```jsonc
{
  "data": {
    "address": "lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu",
    "publicKey": "", // Optional
    "name": "genesis_56", // Optional
    "pendingUnlocks": [ // prev: unlocking
      {
        "validatorAddress": "lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu", // prev: delegateAddress
        "amount": "1081560729258",
        "tokenID": "0000000000000000",
        "unstakeHeight": 30, // prev: unvoteHeight
        "expectedUnlockableHeight": 300,
        "unlockable": true
      }
    ]
  },
  "meta": {
    "count": 10,
    "offset": 0,
    "total": 10
  },
}
```

### PR DESCRIPTION
**Pull request by sameersubudhi**
### What was the problem?
This PR resolves #1350 

### How was it solved?
- [x] The endpoint are updated as expected above
- [x] The response body is up-to-date
- [x] All the methods, variables, files are renamed and refactored appropriately
- [x] Swagger is up-to-date
- [x] Integration tests are up-to-date
- [x] Gateway bugs are fixed
- [x] Code cleanup

### How was it tested?
Local against Lisk Core v4.0.0-alpha.5

### CODE REVIEW

File: services/gateway/apis/http-version3/swagger/definitions/pos.json [hunk 1]
```diff
@@ -107,74 +107,92 @@
             }
         }
     },
-    "unlock": {
+    "PoSUnlock": {
```
Comment (ManuGowda) [inline services/gateway/apis/http-version3/swagger/definitions/pos.json:110]: what case are we following? it should be posUnlock?

File: services/gateway/shared/regex.js [hunk 2]
```diff
@@ -18,8 +18,8 @@ const ADDRESS_LISK32 = /^lsk[a-hjkm-z2-9]{38}$/;
 const NONCE = /^[0-9]+$/;
 const TIMESTAMP_RANGE = /^(?:(?:\d+)|(?::(?:\d+))|(?:(?:\d+):(?:\d+)?))$/;
 const HEIGHT_RANGE = /^(?:(?:\d+)|(?::(?:\d+))|(?:(?:\d+):(?:\d+)?))$/;
-const NAME = /^[\w!@$&.]{1,20}$/;
-const NAME_CSV = /^[\w!@$&.,]{1,}$/;
+const NAME = /^[\w!@$&.]{3,20}$/;
```
Comment (ManuGowda) [inline services/gateway/shared/regex.js:21]: why the regex change?

File: services/gateway/apis/http-version3/swagger/definitions/pos.json (hunk 1 above)
Comment (sameersubudhi) [reply services/gateway/apis/http-version3/swagger/definitions/pos.json]: We are using pascal case to define swagger param definitions everywhere. Currently updated the ones concerning this PR but will gradually update the remaining.

File: services/gateway/shared/regex.js (hunk 2 above)
Comment (sameersubudhi) [reply services/gateway/shared/regex.js:21]: The regex was updated to integrate the restrictions we had on the gateway: https://github.com/LiskHQ/lisk-service/pull/1368/files#diff-0035135774944e209dca330a711431d95be645382b8ccf059ac27ed2535b60adL27-L29

**Pull request merged**